pylint/checkers/typecheck.py

```diff
@@ -1567,6 +1551,34 @@ def _check_invalid_sequence_index(self, subscript: nodes.Subscript):
         self.add_message("invalid-sequence-index", node=subscript)
         return None
 
+    def _check_not_callable(
+        self, node: nodes.Call, inferred_call: Optional[nodes.NodeNG]
+    ) -> None:
+        """Checks to see if the not-callable message should be emitted
+
+        Only functions, generators and objects defining __call__ are "callable"
+        We ignore instances of descriptors since astroid cannot properly handle them yet
+        """
+        if inferred_call and not inferred_call.callable():
+            if isinstance(inferred_call, astroid.Instance):
+                # Don't emit if we can't make sure this object is callable.
+                if not has_known_bases(inferred_call):
+                    return
+
+                if inferred_call.parent and isinstance(
+                    inferred_call.scope(), nodes.ClassDef
+                ):
+                    # Ignore descriptor instances
+                    if "__get__" in inferred_call.locals:
+                        return
+                    # NamedTuple instances are callable
+                    if inferred_call.qname() == "typing.NamedTuple":
+                        return
+
+            self.add_message("not-callable", node=node, args=node.func.as_string())
+        else:
+            self._check_uninferable_call(node)
```

Comment:
```suggestion
if not inferred_call or inferred_call.callable():
    return self._check_uninferable_call(node)
if not isinstance(inferred_call, astroid.Instance):
    return
# Don't emit if we can't make sure this object is callable.
if not has_known_bases(inferred_call):
    return
if inferred_call.parent and isinstance(
        inferred_call.scope(), nodes.ClassDef
):
    # Ignore descriptor instances
    if "__get__" in inferred_call.locals:
        return
    # NamedTuple instances are callable
    if inferred_call.qname() == "typing.NamedTuple":
        return
self.add_message("not-callable", node=node, args=node.func.as_string())
```
One step further :)

Reply:
Slightly adapted this suggestion and added some extra white lines + comments in latest commit.